src/FastSerialization/FastSerialization.cs

```diff
@@ -174,6 +174,10 @@ interface IStreamReader : IDisposable
         /// </summary>
         string ReadString();
         /// <summary>
+        /// Read a span of bytes from the stream.
+        /// </summary>
+        void Read(Span<byte> span);
```

Comment:
Let me know how you want to handle it.   As mentioned, I am OK with your approach above.   Split out the new Mapped Reader/writers into their own file(s), and make sure that Read() works if the size of the target read is larger than the buffer size of the reader/writer (there is a loop until done).   After that I am OK with it.

Reply:
> An alternative is `Span<byte> GetData(int length)`, where the stream still owns the buffer and the returned Span is only valid until the next operation on the Stream.

I'm not a fan of this approach because the usable life of the `Span<byte>` is difficult to work with. It seems more likely to be a source of bugs without a clear performance need.

Reply:
I am OK with Read(Span).